extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -467,10 +515,25 @@ public boolean canRestore()
   public void stopGracefully()
```

Comment:
This doesn't look like it does what a `stopGracefully` should do. Maybe we should rename that method to something like `suspend` (not in this PR though). But, what it's supposed to do is suspend a task quickly so the jvm can be shut down and then restarted. Specifically it shouldn't be trying to publish or anything like that – just persist to disk and then stop.